### ISSUE TITLE
`intent_slot_classification_model` tutorial wont converge

### ISSUE THREAD

**Issue by david-waterworth** [NVIDIA/NeMo#6808]
I'm trying to run the `intent_slot_classification_model` and I'm not getting anywhere near the accuracy that the workbook implies I should. I'm running the tutorial from `Joint_Intent_and_Slot_Classification.ipynb`

The workbook contains the following markedown

```
After training for 5 epochs, which should take no more than few minutes, you can expect training precision for this data set to be around these numbers (the accuracy will gradually continue to improve for this dataset up to about 50 epochs of training): 

Intents:

    micro avg                                               85.04      85.04      85.04       1076
    macro avg                                               81.13      80.81      79.36       1076
    weighted avg                                            84.10      85.04      83.54       1076
    
Slots:

    -------------------
    micro avg                                               89.10      89.10      89.10       7199
    macro avg                                               21.86      18.56      18.18       7199
    weighted avg                                            84.42      89.10      86.01       7199
```

But I'm not able to reproduce this, the training / eval loss is stuck around 4.12 and the precision/recall/f1 is really low (less than 2%) - I don't actually get any improvement after the first epoch.

I've tried on bare-metal and aws sagemaker with the exact same results.

**Expected behavior**

Expect much better precision/recall when running tutorial workbook

**Environment overview (please complete the following information)**

 - Environment location: AWS Sagemaker (Pytorch 2.0.0 Python 3.10 GPU Optimised)
 - Method of NeMo install: [pip install or from source]. Ran the install command from the workbook

```
BRANCH = 'stable'
!python -m pip install git+https://github.com/NVIDIA/NeMo.git@$BRANCH#egg=nemo_toolkit[nlp]
```

 - If method of install is [Docker], provide `docker pull` & `docker run` commands used

**Environment details**

AWS Sagemaker (Pytorch 2.0.0 Python 3.10 GPU Optimised)

(same results running on bare metal with torch 2.0.1 with 3090 RTX GPU)

Note I don't have Apex installed, I suspect perhaps I'm using a newer version of lightning that the one that was used when the notebook was created but not really sure?

**Comment by Rumeysakeskin** [comment]
Hello @david-waterworth ,
I am facing the same problem.  Did you come up with any solution?

**Comment by david-waterworth** [comment]
@Rumeysakeskin yes you're right, that version does work. As far as I can see it using the model model but a different training script.

It didn't work on Sagemaker or my local machine though, implying that the issue is down to the specific package versions installed. On my local machine I was getting numpy errors so I manually downgraded numpy and numba 

```
pip install numpy==1.22.4
pip install numba==0.56.4
```

and now can run locally - I'll create a separate issue for this

**Comment by Rumeysakeskin** [comment]
Hello @david-waterworth ,
I use  Dockerfile which extends the PyTorch22.04-py3 NGC container and encapsulates some dependencies.
Thus I didn't encounter any dependency conflict issues. The dependencies you need for the Nvidia NeMo model are provided here.
To create your own container, you can choose a PyTorch container from [NVIDIA PyTorch Container Versions](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-22-11.html#rel-22-11) according to your cuda and ubuntu version.
For example I have `cuda:11.6.2` and `ubuntu:20.04`
My Dockerfile is as follows:
```python 
FROM nvcr.io/nvidia/pytorch:22.04-py3
WORKDIR /path/to/working/directory/
COPY requirements.txt .
RUN pip install -r requirements.txt
```
`requirements.txt`
```python 
inflect
unidecode
torch
numpy
scipy
librosa
cython
matplotlib
fastapi
hydra-core
uvicorn
service_streamer
```

Also since I built it from Jupyter Notebook, I installed the following setups there:
```python
# install NeMo
BRANCH = 'main'
!python -m pip install git+https://github.com/NVIDIA/NeMo.git@$BRANCH#egg=nemo_toolkit[nlp]

!apt-get update
!apt-get install -y libsndfile1 ffmpeg
```

In this way, it works for me. I hope I could help.
